src/libraries/System.Private.Xml/src/System/Xml/Schema/DataTypeImplementation.cs

```diff
@@ -1143,11 +1143,12 @@ internal override RestrictionFlags ValidRestrictionFlags
         Error:
             return exception;
 
-            // TODO: Replace with https://github.com/dotnet/runtime/issues/76478 once available
-            [UnconditionalSuppressMessage("AotAnalysis", "IL3050:AotUnfriendlyApi",
-                Justification = "Array type is always present as it is passed in as a parameter.")]
             static Array ToArray(ArrayList values, Type arrayType)
```

Comment:
This does not need to be in a separate method now that it does not have UnconditionalSuppressMessage annotation
```
if (_itemType.Variety == XmlSchemaDatatypeVariety.Union)
{
...
    arrayType = typeof(XmlAtomicValue[]);
}
else
{
....
    arrayType = _itemType.ListValueType;
}
if (values.Count < _minListSize)
{
    return new XmlSchemaException(SR.Sch_EmptyAttributeValue, string.Empty);
}
Array array = Array.CreateInstanceFromArrayType(arrayType, values.Count);
values.CopyTo(result);
```